File: week2/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // Get text from user
    printf("Text: ");
    char txt[1024];
    scanf("%[^\n]%*c", txt);
    int txtlen = strlen(txt);
    int sentences = 0, words = 1, letters = 0;

    for (int i = 0; i < txtlen; i++)
    {
        if (txt[i] == '.' || txt[i] == '!' || txt[i] == '?')
        {
            sentences++;
        }
        if (txt[i] == ' ')
        {
            words++;
        }
        if (isalpha(txt[i]))
        {
            letters++;
        }
    }

    // Calculate readability
    float L = (float) letters / (float) words * 100.0;
    float S = (float) sentences / (float) words * 100.0;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);
    if (grade < 1 || grade > 16)
    {
        if (grade < 1)
        {
            printf("Before Grade 1\n");
        }
        if (grade > 16)
        {
            printf("Grade 16+\n");
        }
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

File: week6/readability/readability.py
sentences = words = letters = 0
words += 1
txt = input("Text: ")
for c in txt:
    if c in '.!?':
        sentences += 1
    elif c == ' ':
        words += 1
    elif c.isalpha():
        letters += 1
L = letters / words * 100
S = sentences / words * 100
grade = round(0.0588 * L - 0.296 * S - 15.8)
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")
